src/CacheInterface.php

```diff
@@ -120,7 +120,7 @@ public function delete(string $key): PromiseInterface;
      *
      * @param string[] $keys A list of keys that can obtained in a single operation.
      * @param mixed $default Default value to return for keys that do not exist.
-     * @return PromiseInterface<array> Returns a promise which resolves to an `array` of cached values
+     * @return PromiseInterface<array<string,mixed>> Returns a promise which resolves to an `array` of cached values
```

Comment:
Have we considered making the cache typeable? (Out of scope for this PR of course.)

```suggestion
     * @return PromiseInterface<array<string,T>> Returns a promise which resolves to an `array` of cached values
```

Reply:
Not sure how useful this is, but happy to discuss in a separate PR if this is something you want to look into :+1:

Reply:
I will do a PR once this is in together with one for DNS to show that off after doing some more research :+1: .